src/extension.ts

```diff
@@ -46,13 +46,22 @@ let treeDataProvider: Disposable = new Disposable(() => undefined);
 let clients: Map<string, LanguageClient> = new Map();
 let outputChannel: OutputChannel = window.createOutputChannel('gauge');
 let specExplorerActiveFolder: string = "";
+let terminalStack: Terminal[] = [];
 
 export function activate(context: ExtensionContext) {
+    let currentExtensionVersion = extensions.getExtension(GAUGE_EXTENSION_ID)!.packageJSON.version;
+    let hasUpgraded = hasExtensionUpdated(context, currentExtensionVersion);
+    context.subscriptions.push(new WelcomePageProvider(context, hasUpgraded));
+
+    context.subscriptions.push(commands.registerCommand
```

Comment:
Consider moving the command + registration to a separate class. Even `terminalStack` can be encapsulated in the new class. See `WelcomeProvider` for example.